pkg/standalone-utils/contracts/relayer/BaseRelayerLibrary.sol

```diff
@@ -149,7 +164,13 @@ contract BaseRelayerLibrary is IBaseRelayerLibrary {
         // solhint-disable-next-line no-inline-assembly
         assembly {
             value := sload(slot)
-            sstore(slot, 0)
+        }
+
+        if (_isTemporaryChainedReference(ref)) {
```

Comment:
Some thoughts about naming this change.

We could invert the checks and search for 'read-only' prefixes instead of 'temporary' ones. But then we would need to either negate the result, or compare to `false` in this `if` clause. IMO it's more readable like this.